services/app-api/handlers/kafka/get/forceKafkaSync.js

```diff
@@ -78,34 +64,25 @@ export const main = handler(async (event, context) => {
 
   for (const tableName of tableNames) {
     console.log(`Starting to scan table ${tableName}`);
-    let startingKey;
+    let data;
 
-    let keepSearching = true;
-    // Looping to perform complete scan of tables due to 1 mb limit per iteration
-    while (keepSearching) {
-      let data;
-
-      try {
-        [startingKey, keepSearching, data] = await scanTable(
-          tableName,
-          startingKey,
-          keepSearching
-        );
-      } catch (err) {
-        console.error(`Database scan failed for the table ${tableName}
-                     with startingKey ${startingKey} and the keepSearching flag is ${keepSearching}.
-                     Error: ${err}`);
-        throw err;
-      }
+    try {
+      data = await dynamoDb.scan({
+        TableName: tableName,
+      });
```

Comment:
This is actually the one case (across all of our apps, as far as I know) where the scan pagination was relevant. The `mergeLastSynced` operation can run on each page of results while we wait for the next, and processed objects can be garbage collected as the data streams through this lambda. And, if I recall correctly, SEDS has a _ton_ of data in dynamo, to the point where that might start to matter.

If we want to preserve those performance characteristics... the easiest thing would be, basically, to leave this file alone. Perhaps the legacy `scan` function would remain in dynamodb-lib for now, under the scarier name `scanSome`?

I'd love to be able to use the SDKv3 `paginate` here, but it's locked behind a somewhat intimidating .tsconfig change.

Reply:
The old `scan` would have performed a full scan as well, despite the pagination logic expected in `forceKafkaSync`. We will keep the same full scan logic for now, knowing it can be improved later.
https://github.com/Enterprise-CMCS/macpro-mdct-seds/blob/master/services/app-api/libs/dynamodb-lib.js#L25-L35